### ISSUE TITLE
How to pad mini-batches in a training loop?

### ISSUE THREAD

**Issue by trpstra** [LaurentMazare/tch-rs#599]
First of all, thanks for this wonderful library!

I have been reading through the tests and examples but I'm struggling to figure out one basic use case, so I hope you don't mind me asking here. What is the idiomatic way to iterate through a labeled dataset and pad the mini-batches before each step? I understand `tch::data::Iter2` provides something akin to pytorch data loaders but the `Iter2` constructor requires me to input the data as tensors, which means padding all data samples to the maximum length. As I'm working with language data this is not feasible in my case.
I looked at the translation example but it doesn't seem to use batching. Is my best option to implement something like `tch::data::TextData{Iter}` for labeled data myself, or have I missed some essential piece of the API somewhere?

Thanks a lot.

**Comment by LaurentMazare** [comment]
Hey, I don't think there is anything like this baked in `tch` for now. Might be good to look at [rust-bert](https://github.com/guillaume-be/rust-bert) if you haven't done so already as it's likely to have more specialized NLP code.

**Comment by trpstra** [comment]
Thanks, I had a look at rust-bert but unless I am mistaken it doesn't seem to support model training, only running inference on pretrained huggingface models. 
I may have a go at implementing it myself based on the `tch::data::TextData` code.
Cheers!
